### PR TITLE
Implementation of issue #346

### PR DESCRIPTION
**Pull request by oliverschaefer**
See issue #346 for details.

### CODE REVIEW

**Comment by YuanZhu-apple** [comment]
Thanks for this!
Can you also add a piece of demo code to ORKTest app?

**Comment by oliverschaefer** [comment]
@YuanZhu-apple Done.

File: ResearchKit/Common/ORKTaskViewController.m [hunk 1]
```diff
@@ -981,6 +981,10 @@ - (void)collectToolbarItemsFromViewController:(UIViewController *)viewController
     _pageViewController.toolbarItems = viewController.toolbarItems;
     _pageViewController.navigationItem.leftBarButtonItem = viewController.navigationItem.leftBarButtonItem;
     _pageViewController.navigationItem.rightBarButtonItem = viewController.navigationItem.rightBarButtonItem;
+    if (!_showsProgressInNavigationBar) {
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKTaskViewController.m]: `progressLabel` determined by two factors:
1. `_showsProgressInNavigationBar`
2. ` [_task respondsToSelector:@selector(progressOfCurrentStep:withResult:)]`

```
Line 910
if (self.showsProgressInNavigationBar &&  [_task respondsToSelector:@selector(progressOfCurrentStep:withResult:)]) {
    ...
}
```

Can you refactor this two factors into a helper method `shouldDisplayProgressLabel` and use it here and `line 910` ?

**Comment by oliverschaefer** [comment]
@YuanZhu-apple Done.

**Comment by YuanZhu-apple** [comment]
Thanks @oliverschaefer 
r=me. 
@umerkhan-apple  Can you help me double check this PR?

**Comment by umerkhan-apple** [comment]
r=m3

**Pull request merged**